libs/blocks/global-navigation/global-navigation.js

```diff
@@ -373,6 +479,117 @@ class Gnav {
+      const children = [CONFIG.universalNav.components.profile];
+      // reset sign up value on change
+      children[0].attributes.isSignUpRequired = false;
+
+      this.universalNavComponents?.forEach((component) => {
+        if (component === 'profile') return;
+        if (component === 'signup') {
+          children[0].attributes.isSignUpRequired = true;
+          return;
+        }
+
+        children.push(CONFIG.universalNav.components[component]);
+      });
+
+      return children;
+    };
+
+    const onAnalyticsEvent = (data) => {
+      if (!data) return;
+
+      const getInteraction = () => {
+        const {
+          event: { type, subtype } = {},
+          source: { name } = {},
+          content: { name: contentName } = {},
+        } = data;
+
+        switch (`${name}|${type}|${subtype}|${contentName || ''}`) {
+          case 'profile|click|sign-in|':
+            return `Sign In|gnav|${experienceName}|unav`;
+          case 'profile|render|component|':
+            return `Account|gnav|${experienceName}`;
+          case 'profile|click|account|':
+            return `View Account|gnav|${experienceName}`;
+          case 'profile|click|sign-out|':
+            return `Sign Out|gnav|${experienceName}|unav`;
+          case 'app-switcher|render|component|':
+            return 'AppLauncher.appIconToggle';
+          case `app-switcher|click|app|${contentName}`:
+            return `AppLauncher.appClick.${convertToPascalCase(contentName)}`;
+          case 'app-switcher|click|footer|adobe-home':
+            return 'AppLauncher.adobe.com';
+          case 'app-switcher|click|footer|all-apps':
+            return 'AppLauncher.allapps';
+          case 'app-switcher|click|footer|adobe-dot-com':
+            return 'AppLauncher.adobe.com';
+          case 'app-switcher|click|footer|see-all-apps':
+            return 'AppLauncher.allapps';
+            // TODO: add support for notifications
+          default:
+            return null;
+        }
+      };
+      const interaction = getInteraction();
+
+      if (!interaction) return;
+      // eslint-disable-next-line no-underscore-dangle
+      window._satellite?.track('event', {
+        xdm: {},
+        data: { web: { webInteraction: { name: interaction } } },
+      });
+    };
+
+    const getConfiguration = () => ({
+      target: this.blocks.universalNav,
+      env: environment,
+      locale,
+      imsClientId: window.adobeid?.client_id,
+      theme: 'light',
+      analyticsContext: {
+        consumer: {
+          name: 'adobecom',
+          version: '1.0',
+          platform: 'Web',
+          device: getDevice(),
+          os_version: navigator.platform,
+        },
+        event: { visitor_guid: visitorGuid },
+        onAnalyticsEvent,
+      },
+      children: getChildren(),
+    });
+
+    window.UniversalNav(getConfiguration());
+
+    isDesktop.addEventListener('change', () => {
+      window.UniversalNav.reload(getConfiguration());
```

Comment:
Do we need to expose this in case a consumer would want to reload anything?

Reply:
There's no business case for this, but if we were to implement this, we'd do it through events. For now we don't need to expose this.